### DISCUSSION TITLE
Work Being Done to Update the Project

### DISCUSSION THREAD

**Discussion by jprochazka** [jprochazka/adsb-receiver#542] [Announcements]
After a long time away from the project I have begun to update the scripts to work with the most recent releases made since the last time I was able to work on the project. My first goal is to bring the current code up to date. This means reworking the scripts to work with currently supported distributions of Linux as well as make things work with the most current versions of the third-party code bases it interacts with as well as removing support for now defunct packages. These changes can be tracked in a new branch named "[phoenix](https://github.com/jprochazka/adsb-receiver/tree/phoenix)", get it? This branch also has the following [milestone](https://github.com/jprochazka/adsb-receiver/milestone/16) attached to it. Once completed this branch will become v2.8.0. For this version do not expect to see anything new. Once the current code is up and working, I am planning to begin work on the portal. Maybe some fancier graphs and a more modernized layout based on more modern technologies?

Something to take note of is the fact support for Ubuntu will be lacking at this time. I am currently working on FlightAware support at the moment and their code does not play nice with the most recent versions of the distribution. This is to be expected being their software is geared mainly towards Raspberry PI OS. In the future I hope to once again get things working on other Debian based distribution such as Ubuntu but in order to get things up and running as quickly as possible Debian and Raspberry Pi OS compatibility are going to be my main focus.

It should be noted that at some point in the near future, I will be going through the issues and doing a bit of housekeeping. Over time I will be deleting many of the older issues which are addressed in the upcoming version in order to keep the issue tracker saine and of use to those who may experience issues from here on out. I also merged a few pull requests into the master branch which people were kind enough to submit during my absence. Thanks to those who contributed!

As for documentation, I will be relaunching adsbreceiver.net shortly. This site will contain new updated documentation I am putting together as I test the different installation operations during the update process.

_**Update** June 20th, 2024_

There are only a handful of items left which need to be addressed with the lite version of the portal as well as a couple broken graphs after which updating the advanced version will be done. Once the portal is ready, I think I will feel comfortable enough to merge the changes into the master branch and create a new release.

The issue tracker has been cleared out. Due to the out-of-date state of the master branch, I ask that issues only be opened pertaining to the new phoenix branch until it has been merged into master. Any issues related to the current master branch will be disregarded at this time.

Live testing should be done on distributions listed as not tested in issues #540, #541, #543, #545, #546, #548, #553, and #554 before they are closed. If anyone wants to give them a once over and report their results feel free to do so.

Here is the CHANGELOG for the phoenix branch as of today.

Installers

- The dump1090-fa and piaware_builder install process has been updated to support the latest versions.
- The dump978 script has been rewritten in order to use FlightAwares version of dump978.
- The Beast Splitter script has been updated to work with the current version of Beast Splitter.
- The Flightradar24 script has been updated to work with the current install script.
- The Plane Finder script has been updated to work with current client releases.
- Removed the unattended install option for the time being.
- Removed the option to install dump1090-mutability.
- Removed the option to install dump1090 HPTOA.
- Removed the option to install AboveTustin.
- Removed the option to install ADSBHub support due to the fact it was incomplete.
- Removed incomplete RTL-SDR OGN setup scripting.
- Version check for dump1090-fa has been fixed.
- Fix bug where PiAware was installed every run even if already installed.

Portal

- All portal related Python scripts have been updated to work with Python 3.
- TinyMCE has been replace by CKeditor in the portal.

**Comment by laferrierem** [comment]
Good to see you back and good to see the project has been revived and
updated. I still have two receivers running your earlier code. They are
update starved and I look forward to helping with testing once things get
going again.

Welcome back!

**Reply by jprochazka** [reply]
Hello @laferrierem thanks for suffering through the downtime. Is there anything you would like to see set at a higher priority on my TODO list as far as what you have installed currently? Meaning decoders, feeders, addons, etc.

**Comment by dranreb38** [comment]
Ha I'm happy about the birth of the phoenix project, I can't wait to help with testing once things are finished.

Happy to see you again!
https://www.flightaware.com/adsb/stats/user/dranreb38#stats-59352

**Comment by Tom-Neverwinter** [comment]
Welcome back!
